lib/brain.js

```diff
@@ -2576,11 +2578,17 @@ Brain.prototype._physicalDispense = function _physicalDispense () {
 
       this._transitionState('fiatComplete', {tx: displayTx})
       // this.waitForCollection()
-    })
-    .catch(err => {
-      console.log(err.stack)
+    }, err => {
       emit('billDispenserCollected')
-      return this._fiatError(err)
+
+      this.fastUpdateTx({
+        error: err.message,
+        errorCode: err.statusCode || null
+      })
+
+      // bounce the error to be catched
+      // by _dispense function
+      throw err
```

Comment:
the state change is handled externally in _dispense

Reply:
👍